File: 프로그래머스/1/12982. 예산/예산.java
import java.util.Arrays;

class Solution {
    public int solution(int[] d, int budget) {
        int answer = 0;
        Arrays.sort(d);
        
        for(int i: d){
            if(i>budget){
                break;
            }else{
                budget-=i;
                answer++;
            }
        }
        return answer;
    }
}

File: 프로그래머스/1/12982. 예산/예산.js
function solution(d, budget) {
    var answer = 0;
    d.sort((a, b)=>a-b)
    
    let bClone = budget;
    for(let i of d){
        if(i<=bClone){
            bClone-=i;
            answer++;
            continue;
        }
        break;
    }
    
    return answer;
}